### PR TITLE
Handle room collapsed

### PR DESCRIPTION
**Pull request by cyberblast**
only for owned rooms. assumed to be collapsed if no worker and no hauler and no pioneers present in that room. 

Actions: 
* flush high spawn queue
* try to spawn a worker
* set a pioneer flag

#517

### CODE REVIEW

**Comment by torin555** [comment]
Does this address my second comment on #517?

**Comment by cyberblast** [comment]
yes. for all pioneers.

File: task.pioneer.js [hunk 1]
```diff
@@ -13,7 +13,32 @@ mod.register = () => {
...
     // a creep died
     Creep.died.on( name => Task.pioneer.handleCreepDied(name) );
+    // a room collapsed
+    Room.collapsed.on( room => Task.pioneer.handleRoomDied(room) );
 };
+mod.handleRoomDied = room => {
+    // try to spawn a worker
+    if( room.energyAvailable > 199 ) {
+        // flush high queue
+        while(room.spawnQueueHigh.length > 0 ) room.spawnQueueHigh.pop();
```
Comment (karlthepagan) [inline task.pioneer.js]: `room.spawnQueueHigh.splice(0,room.spawnQueueHigh.length)`

Comment (cyberblast) [inline task.pioneer.js]: ok

**Comment by cyberblast** [comment]
Obsolete since #570

**Comment by cyberblast** [comment]
oops wrong PR 😆

**Comment by cyberblast** [comment]
Yes, I had this in mind as well. As somehow "shit happens" and there are stale entries blocking the "workflow". 
Maybe something which could get called regularly like every 500 ticks...

**Pull request merged**